src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -176,15 +225,33 @@ class ArnoldOutput : public IECore::RefCounted
 				AiNodeSetStr( m_driver.get(), AiParamGetName( fileNameParameter ), output->getName().c_str() );
 			}
 
+			IECore::StringVectorDataPtr customAttributesData = new IECore::StringVectorData();
+			std::vector<std::string> &customAttributes = customAttributesData->writable();
 			for( IECore::CompoundDataMap::const_iterator it = output->parameters().begin(), eIt = output->parameters().end(); it != eIt; ++it )
 			{
 				if( boost::starts_with( it->first.string(), "filter" ) )
 				{
 					continue;
 				}
+
+				if( boost::starts_with( it->first.string(), "header:" ) )
+				{
+					std::string formattedString = formatHeaderParameter( it->first.string().substr( 7 ), it->second.get() );
+					if( !formattedString.empty())
+					{
+						customAttributes.push_back( formattedString );
+					}
+					continue;
+				}
+
 				ParameterAlgo::setParameter( m_driver.get(), it->first.c_str(), it->second.get() );
 			}
 
+			if( AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( m_driver.get() ), "custom_attributes" ) )
+			{
+				ParameterAlgo::setParameter( m_driver.get(), "custom_attributes", customAttributesData.get() );
```

Comment:
Can we not just support both? By initialising the array with any custom_attributes value we find in the parameters, and then appending the "header:" values?